R/backend-mssql.R

```diff
@@ -333,6 +335,7 @@ simulate_mssql <- function(version = "15.0") {
       second = function(x) sql_expr(DATEPART(SECOND, !!x)),
 
       month = function(x, label = FALSE, abbr = TRUE) {
+        check_bool(label)
```

Comment:
I'd be tempted to move the check for `check_unsupported_arg()` next to this check.

Reply:
Oh, interesting. I moved that `check_unsupported_arg()` call to be evaluated unconditionally before realizing that the only allowed value for `abbr` is `FALSE` and the default is `TRUE`. Bringing the check out of `if (label)` would mean that `month(x)` would fail by default on SQL server. I opted not to make your suggested change here and will leave this as unresolved for a day or two to give you a chance to reply before merging.